src/network/networkd-address.c

```diff
@@ -551,6 +551,9 @@ static bool link_is_static_address_configured(Link *link, Address *address) {
                 else if (address->family == AF_INET6 && net_address->family == AF_INET6 &&
                          in_addr_equal(AF_INET6, &address->in_addr, &net_address->in_addr_peer) > 0)
                         return true;
+                else if (address->family == AF_INET && net_address->family == AF_INET &&
+                         in_addr_equal(AF_INET, &address->in_addr, &net_address->in_addr) > 0)
```

Comment:
I don't understand this comparison. We check in_addr_peer for AF_INET6 and in_addr for AF_INET... So I think this at least needs a comment in the code why `address_equal()` doesn't suffice and why we are comparing different fields for different protocols.

Reply:
I added a comment.

Reply:
Thanks. This helps a lot. I still don't understand the second `else if` though:
you say that "Address stored in Link does not [contain the peer address]", so I don't understand why we're comparing the two.
`address_equal() calls address_compare_func() which does `memcmp(&a1->in_addr, &a2->in_addr)`. Is this supposed to return a different result than `in_addr_equal()`?

Reply:
Not only `in_addr` element, but `address_compare_func()` also compares prefix generated by `address_prefix()` it may return different value if a peer address is set.

The comment in the code is revised.